Uploaded lab5 python file and README.md description

sofiaperiv opened the pull request:
Lab5 README description and python file have been uploaded

sofiaperiv committed:
Lab5 description added README.md

sofiaperiv committed:
Lab5.py has been uploaded

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+    def __del__(self):
+        print("Cleanup has been completed")
+        
+    def getAll(self):
```
```suggestion
    def get_all_parameters(self):

```

sofiaperiv replied:
Thanks, I've changed this method name to the suggested one, which is more suitable according to PEP8.

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+        
+    def getAll(self):
+        """
+        Explanatiom huy na blude
```
```suggestion
        Explanation of all parameters

```

sofiaperiv replied:
Thanks for noticing it. I will not commit your suggestion as I reckon it will be better to remove this docstring altogether

OlhaLeno replied:
I completely agree with you

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+    def getAll(self):
+        """
+        Explanatiom huy na blude
+        :return: pizdec
```
```suggestion
        :return: self parameters

```

sofiaperiv replied:
Thanks for noticing it. I will not commit your suggestion as I reckon it will be better to remove this docstring altogether

OlhaLeno replied:
I completely agree with you.

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+        self.weather_type = weather_type
+        
+    def __del__(self):
+        print("Cleanup has been completed")
```
```suggestion
        print("Clean up has been completed")

```

sofiaperiv replied:
"Cleanup" is a noun and "Clean up" is a verb. In the given statement I used a noun, hence suggested change has been rejected

OlhaLeno replied:
Oh, thanks for explaining and sorry for correcting.

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+    weather1 = Weather(1, City.LVIV, "Ukraine", 2, 80, 10, WeatherType.FOGGY)
+    weather2 = Weather(9, City.KYIV, "Ukraine", 4, 67, 7, WeatherType.CLOUDY)
+    weather3 = Weather(3, City.BURSHTYN, "Ukraine", 3, 72, 8, WeatherType.SUNNY)
+    weather4 = Weather(6, City.LVIV, "Ukraine", 7, 89, 9, WeatherType.RAINY)
```
```suggestion
    weather4 = Weather(6, City.Dnipro, "Ukraine", 7, 89, 9, WeatherType.RAINY)

```

sofiaperiv replied:
I didn't commit your suggestion since I must have an example of the outcome "The typical weather in Lviv", but I added new city Dnipro to another weather instance.

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+    calendar.add_weather_objects(weather4)
+    calendar.add_weather_objects(weather5)
+
+    print("weather1 getAll: ", weather1.getAll())
+    print("weather2 getAll: ", weather2.getAll())
+    print("weather3 getAll: ", weather3.getAll())
+    print("weather4 getAll: ", weather4.getAll())
+    print("weather5 getAll: ", weather5.getAll())
```
```suggestion
    print("weather1 getAll: ", weather1.get_all_parameters())
    print("weather2 getAll: ", weather2.get_all_parameters())
    print("weather3 getAll: ", weather3.get_all_parameters())
    print("weather4 getAll: ", weather4.get_all_parameters())
    print("weather5 getAll: ", weather5.get_all_parameters())

```

sofiaperiv replied:
Thanks, I've committed you suggestion. Your suggested change is better according to PEP8

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+    def __del__(self):
+        print("Cleanup has been completed")
+        
+    def getAll(self):
```
it's going to be clearer what we're getting

sofiaperiv replied:
Can't understand what you suggest me to do

OlhaLeno replied:
If we write "get_all_parameters" instead of "getAll" , I think it will be clearer what exactly we should get.

sofiaperiv replied:
It has been already suggested and I committed that change. Anyway, thanks

OlhaLeno commented on lab5.py:
```diff
@@ -0,0 +1,103 @@
+    weather1 = Weather(1, City.LVIV, "Ukraine", 2, 80, 10, WeatherType.FOGGY)
+    weather2 = Weather(9, City.KYIV, "Ukraine", 4, 67, 7, WeatherType.CLOUDY)
+    weather3 = Weather(3, City.BURSHTYN, "Ukraine", 3, 72, 8, WeatherType.SUNNY)
+    weather4 = Weather(6, City.LVIV, "Ukraine", 7, 89, 9, WeatherType.RAINY)
```
It is better to change the city in the weather4 to another city

sofiaperiv replied:
I didn't commit your suggestion since I must have an example of the outcome "The typical weather in Lviv", but I added new city Dnipro to another weather instance.

OlhaLeno replied:
I have nothing against it. I consider it's nice idea to add new city Dnipro to another weather instance.

sofiaperiv committed:
getAll to get_all_parameters method name updated in lab5.py

Co-authored-by: Olha <151624725+OlhaLeno@users.noreply.github.com>

sofiaperiv committed:
getAll method calling name has been changed in lab5.py

getAll method calling name has been changed to get_all_parameters in lab5.py

Co-authored-by: Olha <151624725+OlhaLeno@users.noreply.github.com>

sofiaperiv committed:
Removed Weather class docstring, added WeatherType element, renamed methods and objects in lab5.py

Docstring in the "Weather" class has been removed.
Added "DNIPRO = 4" to WeatherType(Enum) class.
Renamed methods according to PEP8 naming conventions.
Renamed weather objects according to city and weather type

sofiaperiv committed:
Too many attributes in class 'Weather ' fixed in lab5.py

In class Weather were 8 attributes including 'self'. According to PEP8 in class shouldn't be more than 5 attributes. Therefore, I grouped a couple of  attributes inside new classes such as 'Location', 'TempAndHumidity', 'WIndAndType'
